en/middleware/xrce_dds.md

```diff
@@ -0,0 +1,303 @@
+While most releases should support a very similar set of messages, to be certain you would need to check the yaml file for your particular release.
+<!-- Jublish the set we use?: https://github.com/PX4/px4_msgs/issues/22 -->
+
+Note that ROS 2/DDS needs to have the _same_ message definitions that were used to create the XRCE-DDS client module in the PX4 Firmware in order to interpret the messages:
+
+- If you're using a main or release version of PX4 you can get these by cloning [PX4/px4_msgs](https://github.com/PX4/px4_msgs) into your workspace (branches are provided for each release).
+- If you're creating your own new topic you will need to update [dds_topics.yaml](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/microdds_client/dds_topics.yaml) and also copy the message definitions out of PX4 source tree into any ROS2 workspace that needs them.
+  The messages must be in a folder named `px4_msgs/msg/`.
+  
+  :::note
+  Technically the messages must be in a folder defined by the `type` prefix in the yaml file:
+  As you can see below, this is `px4_msgs::msg::`:
+  
+  ```yaml
+  - topic: /fmu/out/vehicle_odometry
+    type: px4_msgs::msg::VehicleOdometry
+  ```
+  :::
+
+
+### PX4 ROS2 QoS Settings
+
+PX4 QoS settings for publishers are incompatible with the default QoS settings for ROS2 subscribers.
+So if ROS2 code needs to subscribe to a uORB topic, it will need to use compatible QoS settings.
+One example of which is shown in [ROS 2 User Guide > ROS2 Subscriber QoS Settings]()
+
+PX4 uses the following QoS settings for publishers:
+
+```cpp
+uxrQoS_t qos = {
+  .durability = UXR_DURABILITY_TRANSIENT_LOCAL,
+  .reliability = UXR_RELIABILITY_BEST_EFFORT,
+  .history = UXR_HISTORY_KEEP_LAST,
+  .depth = 0,
+};
+```
+
+PX4 uses the following QoS settings for subscribers:
+
+```cpp
+uxrQoS_t qos = {
+  .durability = UXR_DURABILITY_VOLATILE,
+  .reliability = UXR_RELIABILITY_BEST_EFFORT,
+  .history = UXR_HISTORY_KEEP_LAST,
+  .depth = queue_depth,
+};
+```
+
+ROS2 uses the following QoS settings (by default) for publishers and subscriptions: "keep last" for history with a queue size of 10, "reliable" for reliability, "volatile" for durability, and "system default" for liveliness.
+Deadline, lifespan, and lease durations are also all set to "default".
+<!-- From https://github.com/PX4/PX4-user_guide/pull/2259#discussion_r1099788316 -->
+
+## Helpful Resources
+
+- [ROS2 in PX4: ROS2 in PX4: Technical Details of a Seamless Transition to XRCE-DDS](https://www.youtube.com/watch?v=F5oelooT67E) - Pablo Garrido & Nuno Marques (youtube)
+- [PX4 ROS2 offboard tutorial](https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29) (Github gist - JulianOes)
+- [ROS2 PX4 Offboard Tutorial](https://github.com/Jaeyoung-Lim/px4-offboard/blob/2d784532fd323505ac8a6e53bb70145600d367c4/doc/ROS2_PX4_Offboard_Tutorial.md) (Jaeyoung-Lim).
+
+
+
+<!---
+Some of this might be useful.
```

Comment:
Below here is stuff from the old architecture on uart config. I've left here - but probably would like in a "hardware tutorial".